plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -151,10 +157,12 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
         }
 
         Map<Integer, Long> ndvs = readNdvs(
+                icebergTable,
                 snapshotId,
                 // TODO We don't need NDV information for columns not involved in filters/joins. Engine should provide set of columns
```

Comment:
Regarding this TODO (not related to current PR), the engine does provide the columns required for the query (various connectors save this in their ConnectorTableHandle after applyProjection). Do we try to parse all columns here or only the columns accessed by the query ?